db/postgresql-instance.yml

```diff
@@ -0,0 +1,219 @@
+      amount of free disk space in bytes that will trigger warning, 0 means no alarm
+    Type: Number
+  CloudWatchWarnCPU:
+    Default: 0
+    Description: percent of CPU that will trigger warning, 0 means no alarm
+    Type: Number
+    MinValue: 0
+    MaxValue: 100
+    ConstraintDescription: must be between 0 and 100
+  CloudWatchWarnConnections:
+    Default: 0
+    Description: amount of connections that will trigger warning, 0 means no alarm
+    Type: Number
+  OpsWarnSnsTopic:
+    Default: ""
+    Description: ops warn SNS Topic arn
+    Type: String
+  EngineVersion:
+    Description: PostgreSQL Engine Version 
+    Type: String
+Conditions:
+  UseSnapshot: !Not [!Equals [!Ref DBSnapshot, ""]]
+  CreateDB: !Not [!Equals [!Ref CreateDBName, ""]]
+  NamedInstance: !Not [!Equals [!Ref DBInstance, ""]]
+  CreateCloudWatchWarnDiskSpace: !And
+    - !Not [!Equals [!Ref OpsWarnSnsTopic, ""]]
+    - !Not [!Equals [!Ref CloudWatchWarnDiskSpace, 0]]
+  CreateCloudWatchWarnCPU: !And
+    - !Not [!Equals [!Ref OpsWarnSnsTopic, ""]]
+    - !Not [!Equals [!Ref CloudWatchWarnCPU, 0]]
+  CreateCloudWatchWarnConnections: !And
+    - !Not [!Equals [!Ref OpsWarnSnsTopic, ""]]
+    - !Not [!Equals [!Ref CloudWatchWarnConnections, 0]]
+Resources:
+  RDSSubnetGroup:
+    Type: "AWS::RDS::DBSubnetGroup"
+    Properties:
+      DBSubnetGroupName: !Sub Platform-${EnvironmentType}-${AWS::StackName}-db
+      DBSubnetGroupDescription: !Ref EnvironmentType
+      SubnetIds:
+        - Fn::ImportValue:
+            !Sub ${VPCStack}-Subnet1
+        - Fn::ImportValue:
+            !Sub ${VPCStack}-Subnet2
+  MasterDB:
+    Type: "AWS::RDS::DBInstance"
+    DependsOn: RDSSubnetGroup
+    Properties:
+# CloudFormation cannot update a stack when a custom-named resource requires replacing,
+# so when replacement is required (such as when restoring from snapshot),
+# use a new instance name such as oldname-restoredYYYYMMDD
+# Better yet to self manage tearing down of the existing resource, use a new stack with a different instance name
+      DBInstanceIdentifier: !If [NamedInstance, !Ref DBInstance, !Ref "AWS::NoValue"]
+      DBName: !If [CreateDB, !Ref CreateDBName, !Ref "AWS::NoValue"]
+      AllocatedStorage: !Ref DBAllocatedStorage
+      StorageType: gp2
+      DBInstanceClass: !Ref DBInstanceClass
+      DBSnapshotIdentifier: !If [UseSnapshot, !Ref DBSnapshot, !Ref "AWS::NoValue"]
+      Engine: postgres
+      EngineVersion: !Ref EngineVersion
+      MasterUsername: !If [UseSnapshot, !Ref "AWS::NoValue", !Ref DBUser]
+      MasterUserPassword: !If [UseSnapshot, !Ref "AWS::NoValue", !Ref DBPassword]
+      MultiAZ: false
+      Tags:
+        - Key: Environment
+          Value: !Ref EnvironmentType
+        - Key: "prx:cloudformation:stack-name"
+          Value: !Ref AWS::StackName
+        - Key: "prx:cloudformation:stack-id"
+          Value: !Ref AWS::StackId
+      DBSubnetGroupName: !Sub Platform-${EnvironmentType}-${AWS::StackName}-db
```

Comment:
I believe the `DBSUbnetGroup` [return value](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbsubnet-group.html#aws-resource-rds-dbsubnet-group-return-value) is this name, so you should be able to just do a `!Ref RDSSubnetGroup` to get this value, rather than creating the value in both places

Reply:
Thoughts on this?